index.js

```diff
@@ -10,10 +10,17 @@
 var typeOf = require('kind-of');
 
 module.exports = function isNumber(num) {
+  if (!num && num !== 0) return false;
+
   var type = typeOf(num);
   if (type !== 'number' && type !== 'string') {
     return false;
   }
+  if (type === 'string') {
+    num = num.trim();
+    if (!num.length) return false;
+  }
+
   var n = +num;
```

Comment:
`if (n === 0 && type === 'string' && !num.trim()) return false;` this can never be true. Since `0` will never be a string.